### ISSUE TITLE
Build passing... but failing R CHECK

### ISSUE THREAD

**Issue by dwbapst** [travis-ci/travis-ci#3697]
I've been passing some commits for my package paleotree tonight and once I figured out the new Travis CI support for R, I was pretty pleased as my builds were passing... except they were failing R CHECK.

https://travis-ci.org/dwbapst/paleotree/builds/59001450#L5584

Perhaps this has something to do with me adding `--run-donttest`  to `r_check_args`? Here's my .travis.yml for reference:

```
language: r
sudo: required

# Be strict when checking our package
warnings_are_errors: true

r_check_args: --as-cran --run-donttest 

```
And now I'll mention @craigcitro, @eddelbuettel and @hadley

(Apologies for the previous issue I closed without the community:r label!)

**Comment by craigcitro** [comment]
yes, this is surprising. you're hitting a codepath we just added, which clearly isn't behaving as intended.

you can see [earlier in the logs](https://travis-ci.org/dwbapst/paleotree/builds/59001450#L1459) that we've noticed a failure, and are dumping the logs. however, we're not [failing as expected](https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/r.rb#L151) -- i suspect `sh.failure` needs to tweaked. lemme give it a quick spin.

as a secondary thing, we also probably need some better delimiting around the dumped logs.

and yes, i think only folks from the travis team can add labels.

**Comment by dwbapst** [comment]
Ah, I finally managed to pass R CHECK. By the way, I did discover that the passing-but-failing behavior was not affected when I removed `--run-donttest`, although I think you already intuited that.

(And yes, it seems to be so re: labels; a little confused then by the link given at the top of the travis R logs for posting issues...)

**Comment by dwbapst** [comment]
Yo @craigcitro, I don't know if you got a chance to look at your modifications or not, but I wanted to let you know that the 'passing-but-failing-R-CHECK' was continuing in my most recent build (https://travis-ci.org/dwbapst/paleotree/builds/59802327), here's the end tail of the log:

```
Done
You may want to clean up by 'rm -rf /tmp/Rtmp8FZvXT/Rd2pdf592379ad39c8'
0.01s
$ for name in $(find "${RCHECK_DIR}" -type f -name "*fail"); do echo ">>> Filename: ${name} <<<"; cat ${name}; done
R CMD check failed
$ export RCHECK_DIR=$(Rscript -e 'cat(paste0(devtools::as.package(".")$package, ".Rcheck"))')
0.01s
$ grep -q -R "WARNING" "${RCHECK_DIR}/00check.log"; RETVAL=$?
The command "grep -q -R "WARNING" "${RCHECK_DIR}/00check.log"; RETVAL=$?" exited with 0.
Done. Your build exited with 0.
```

**Comment by jennybc** [comment]
:+1: I am also experiencing this: R CMD check fails, but Travis tells me the build is passing. It seems that `warnings_are_errors: true` has the effect that **only** warnings are detected as errors. Errors are, well, ignored? Here's an example in branch where I am experimenting with `.travis.yml` settings:

https://travis-ci.org/jennybc/cellranger/builds/67770424

I've got a test in there asserting that 1 equals 2, so it WILL fail. And it does but because I've got `warnings_are_errors: true` and there were no warnings, Travis reports success.

**Comment by craigcitro** [comment]
quick note from the linked issue -- i know what's going wrong here, but don't know why. fix hopefully coming shortly after understanding. :grinning:

**Comment by krlmlr** [comment]
Perhaps we could add a stupid `sh.cmd 'false'` after [this line](https://github.com/travis-ci/travis-build/blob/e9d5660b1fcee44626ab6f295b399333b4fb6640/lib/travis/build/script/r.rb#L151)?

**Comment by craigcitro** [comment]
so this *should* be fixed now -- @dwbapst if you can easily re-run a build that failed, that would be awesome.

**Comment by dwbapst** [comment]
Hello, apologies for the delay, I was off enjoying Colter Bay with my family.

@craigcitro I'm still a bit of a git/travis novice so I'm less than clear how I'd get Travis to re-run an old broken build without reverting (but maybe that's what you meant), so I just decided to break the build I have with a bad example instead.

Everything seems to run correct now: builds with a bad example fail and those without pass accordingly. 

I unfortunately can't test whether my donttest examples fail R CHECK when `--run-donttest`, as my donttest examples unfortunately timeout after the default 10 min on Travis (which is an entirely separate issue that is still open, I guess: https://github.com/travis-ci/travis-ci/issues/3849 ?).

FYI, I discovered that contrary to this wiki page I found on the r-travis repo (
https://github.com/craigcitro/r-travis/wiki/Porting-to-native-R-support-in-Travis), that sudo is still required (I thought this had perhaps changed since I last tried direct support for R), so I've edited the wiki page accordingly.

**Comment by craigcitro** [comment]
awesome, thanks @dwbapst for trying it out! and thanks a ton for editing the wiki! i'd totally spaced on adding the `sudo` bit there. and yes, the timeout issue is still open.

should you need it: if you go back to an old build log in the travis UI, there should be a "refresh" button in the upper right that will re-run a build.

@BanzaiMan i think this one's safe to close.

**Comment by krlmlr** [comment]
I just had [another successful build](https://travis-ci.org/krlmlr/r-appveyor/builds/109303568#L1973) where R CMD check threw an error. Safest option: grep the logs for ERROR?

**Comment by jimhester** [comment]
@krlmlr That was a bug recently introduced in the native travis, fixed in https://github.com/travis-ci/travis-build/pull/630, I will try to get the travis folks to merge and deploy it today.
